### PR TITLE
fix: cannot destructure property 'date' of 'bisect(...)' as it is und…

### LINKED ISSUE #3479: [Bug] TypeError: Cannot destructure property 'date' of 'bisect(...)' as it is undefined.
### How to reproduce?

1. Search any token, and enter tradeview 
2. Hover on the price line

### What happened?

61564.js:17127 Uncaught TypeError: Cannot destructure property 'date' of 'bisect(...)' as it is undefined.
    at SVGSVGElement.<anonymous> (1564.js:17127)
    at SVGSVGElement.<anonymous> (chunk.2213.js:2369)

### Mask Version:

- [x] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [x] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [x] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by lelenei**
closes #3479

### CODE REVIEW

File: packages/maskbook/src/plugins/hooks/useLineChart.ts [hunk 1]
```diff
@@ -127,7 +127,9 @@ export function useLineChart(
                 return data[index]
             }
 
-            const { date, value } = bisect(d3.mouse(this)[0])
+            const v: { date: Date; value: number } | undefined = bisect(d3.mouse(this)[0])
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/hooks/useLineChart.ts:130]: keep DRY by declaring a function?

**Review by Himself65 — COMMENTED** [review]
remove the unused console.log by the way

```diff
Index: packages/maskbook/src/plugins/hooks/useLineChart.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/maskbook/src/plugins/hooks/useLineChart.ts b/packages/maskbook/src/plugins/hooks/useLineChart.ts
--- a/packages/maskbook/src/plugins/hooks/useLineChart.ts	(revision c9ff9c84a89b9b83150d01ff90853ed43f06db1e)
+++ b/packages/maskbook/src/plugins/hooks/useLineChart.ts	(date 1626753900374)
@@ -14,7 +14,6 @@
     const { top, right, bottom, left, width, height } = dimension
     const contentWidth = width - left - right
     const contentHeight = height - top - bottom
-    console.log(color)
     useEffect(() => {
         if (!svgRef.current) return
```

**Review by Himself65 — COMMENTED** [review]
There also has a bug which tooltip in the wrong place

image

**Review by Himself65 — CHANGES_REQUESTED** [review]
```diff
Index: packages/maskbook/src/plugins/hooks/useLineChart.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/maskbook/src/plugins/hooks/useLineChart.ts b/packages/maskbook/src/plugins/hooks/useLineChart.ts
--- a/packages/maskbook/src/plugins/hooks/useLineChart.ts	(revision c9ff9c84a89b9b83150d01ff90853ed43f06db1e)
+++ b/packages/maskbook/src/plugins/hooks/useLineChart.ts	(date 1626756882416)
@@ -14,7 +14,6 @@
     const { top, right, bottom, left, width, height } = dimension
     const contentWidth = width - left - right
     const contentHeight = height - top - bottom
-    console.log(color)
     useEffect(() => {
         if (!svgRef.current) return
 
@@ -120,16 +119,17 @@
         }
 
         // add tooltip
-        d3.select(svgRef.current).on('mousemove', function (event) {
+        d3.select(svgRef.current).on('mousemove', function () {
+            const mx = d3.mouse(this)[0]
+            if (mx > contentWidth) {
+                return
+            }
             const bisect = (mx: any) => {
                 const date = x.invert(mx)
                 const index = d3.bisector<{ date: Date; value: number }, Date>((d) => d.date).left(data, date, 1)
                 return data[index]
             }
-
-            const v: { date: Date; value: number } | undefined = bisect(d3.mouse(this)[0])
-            if (!v) return
-            const { date, value } = v
+            const { date, value } = bisect(mx)
 
             tooltip.attr('transform', `translate(${Number(x(date)) - 18},${y(value)})`).call(
                 callout,
```

this must be better, doesn't it?

**Review by Himself65 — DISMISSED** [review]

**Review by Himself65 — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**